File: github-trend/python/parse_html.py
from sys import displayhook
from typing import Dict, List
from bs4 import BeautifulSoup

def parseContent(pageContent: str, top: int) -> List[Dict]:
    reposInfo=[]
    soup = BeautifulSoup(pageContent, 'html.parser')
    articles = soup.find_all("article", class_='Box-row')
    for article in articles:
        repoInfo={}
        urlAnchor = article.find("h1").find("a")
        if urlAnchor:
            repoInfo["url"] = urlAnchor["href"][1:]
        descrpitonPara = article.find("p")
        if descrpitonPara:
            repoInfo["description"] = descrpitonPara.get_text().strip()
        langSpan = article.find("span",attrs={"itemprop":"programmingLanguage"})
        if langSpan:
            repoInfo["language"] = langSpan.get_text()
        hotAnchors = article.find_all("a",class_='Link--muted d-inline-block mr-3')
        if hotAnchors:
            repoInfo["star"]=hotAnchors[0].get_text().strip()
            repoInfo["fork"]=hotAnchors[1].get_text().strip()
        reposInfo.append(repoInfo)
    return reposInfo[0:top]

File: github-trend/golang/parse_html.go
package main

import (
	"fmt"
	"os"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

func parseContent(pageContent string, top int) []map[string]string {
	doc, err := goquery.NewDocumentFromReader(strings.NewReader(pageContent))
	if err != nil {
		fmt.Printf("Error parsing the html content: %s\n", err)
		os.Exit(1)
	}
	reposInfo := []map[string]string{}
	doc.Find("article.Box-row").Each(func(i int, s *goquery.Selection) {
		info := map[string]string{}
		url, _ := s.Find("h1 > a[data-hydro-click]").Attr("href")
		info["url"] = url[1:]
		reposInfo = append(reposInfo, info)
		description := strings.TrimSpace(s.Find("p").Text())
		info["description"] = description
		language := strings.TrimSpace(s.Find("span[itemprop=programmingLanguage]").Text())
		info["language"] = language
		s.Find("a[class='Link--muted d-inline-block mr-3']").Each(func(i int, s *goquery.Selection) {
			content := strings.TrimSpace(s.Text())
			switch i {
			case 0:
				info["star"] = content
			case 1:
				info["fork"] = content
			default:
			}
		})
	})
	return reposInfo[:top]
}
